dom/buyer.js

```diff
@@ -0,0 +1,57 @@
+const pureFuncs = require("../utlis/function");
+let products = JSON.parse(localStorage.getItem("productArray"));
+let cart = [];
+function createHTMLForObj(obj) {
+  if (!obj) return;
+  let newObj = document.createElement("div");
+  newObj.setAttribute("class", "test");
+  const objProps = Object.keys(obj);
+  objProps.shift();
+  objProps.forEach((prop) => {
+    let objProP = document.createElement("p");
+    let objProPC = document.createTextNode(prop);
+    objProP.appendChild(objProPC);
+    let objProV = document.createElement("p");
+    let objProVC = document.createTextNode(obj[prop]);
+    objProV.appendChild(objProVC);
+    newObj.appendChild(objProP);
+    newObj.appendChild(objProV);
+    let newBtn = document.createElement("button");
+    newBtn.setAttribute("class", "addToCartBtn");
+    let objID = obj.id;
+    newBtn.addEventListener("click", function () {
+      addToCartList(objID, products, cart);
+    });
+  });
+  return newObj;
+}
```

Comment:
The idea behind this function is great, take a piece of data and return a styled DOM element out of it(UI). But here are few notes on the implementation:
-  `if (!obj) return;` this line in not enough for validation, think deeply about what would be received as an input and what errors could it cause and how can we handle it. A better way to think about it is by validating the user inputs before creating a product object and store it in the local storage, with scenario you don't need to validate the input, I guess! 
- What is the point of these ` const objProps = Object.keys(obj);`,  `objProps.shift();`, `objProps.forEach((prop) => {`?
I don't expect that you have a lot of data within the passed object nor you need to loop over the values since you need to make a different UI for each value, product name would be `h2` tag while product image would have `img` tag. So, I suggest is making you code cleaner and more elegant like this:
```js
// creating tags
const productContainer = document.createElement("div");
const productTitle = document.createElement("h2");
.
.
.
const productImage = document.createElement("img");

// appending class names
productTitle.textContent = product.title // would name the parameter product not obj
.
.
.

// appending content
productContainer.setAttribute("class", "product__container");
.
.
.
productImage.setAttribute("class", "product__image");

// appending children 
productContainer.appendChild(productImage);
.
.
.
```
did you catch the pattern?
- Be careful with variables name, you know what `objProPC` stands for but it took me some time to figure it out. Be descriptive with naming and don't use shortcuts, the long name `productCategory` is way better than the short one `proCat`, seriously `proCat` what is this!?